src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                operationName = cOperation.Constructor.Name;
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                operationName = pOperation.Property.Name;
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                operationName = fOperation.Field.Name;
+            }
+            if (operation is IDelegateCreationOperation dOperation)
+            {
+                if (dOperation.Target is IMethodReferenceOperation mrOperation)
+                    operationName = mrOperation.Method.Name;
+            }
+            else if (operation is IEventReferenceOperation eOperation)
+            {
+                operationName = eOperation.Event.Name;
+            }
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SelectRule(attribute.AttributeType),
+                operationName, attribute.PlatformName, attribute.Version.ToString()));
+        }
+
+        private static void AnalyzeInvocationOperation(IOperation operation, OperationAnalysisContext context,
+            ref PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            AttributeData[]? attributes = null;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod.GetAttributes(), iOperation.TargetMethod.ContainingType);
+            }
+            else if (operation is IObjectCreationOperation cOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(cOperation.Constructor.GetAttributes(), cOperation.Constructor.ContainingType);
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(pOperation.Property.GetAttributes(), pOperation.Property.ContainingType);
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(fOperation.Field.GetAttributes(), fOperation.Field.ContainingType);
+            }
+            else if (operation is IDelegateCreationOperation dOperation)
+            {
+                if (dOperation.Target is IMethodReferenceOperation mrOperation)
+                    attributes = FindAllPlatformAttributesApplied(mrOperation.Method.GetAttributes(), mrOperation.Method.ContainingType);
+            }
+            else if (operation is IEventReferenceOperation eOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(eOperation.Event.GetAttributes(), eOperation.Event.ContainingType);
+            }
+
+            if (attributes != null)
```

Comment:
This is redundant check which can be removed.
```suggestion
```

Reply:
It was not redundant, it was not always set before, but now i will use ImmutableArray